File: tests/test_code/php/resolve_correct_class/rcc.php
<?php
function func_1() {
    echo "this should never get called";
}

class Alpha {
    function func_1() {
        echo "alpha func_1";
        $this.func_1();
        $b = new Beta();
        $b->func_2();
    }

    function func_2() {
        echo "alpha func_2";
    }
}

class Beta {
    function func_1() {
        echo "beta func_1";
        $al = new Alpha();
        $al->func_2();
    }

    function func_2() {
        echo "beta func_2";
    }
}
?>

File: tests/test_code/rb/resolve_correct_class/rcc.rb
def func_1
    puts "this should never get called"
end

class Alpha
    def func_1()
        puts "alpha func_1"
        self.func_1()
        func_1()
        b = Beta.new()
        b.func_2()
    end

    def func_2()
        puts "alpha func_2"
    end
end

class Beta
    def func_1()
        puts "beta func_1"
        al = Alpha.new()
        al.func_2()
    end

    def func_2()
        puts "beta func_2"
    end
end
